From b09a6b53d9580b02f0b3d5f7aaa5c8977f23e8bc Mon Sep 17 00:00:00 2001
From: Teo Hall <teo.hall@freescale.com>
Date: Fri, 30 Oct 2015 08:57:09 -0500
Subject: [PATCH 0867/5242] MLK-11262-1: ARM: imx: add busfreq offset for M4

commit  bc00c2e6070807f000580effdf57c173149364d0 from
https://source.codeaurora.org/external/imx/linux-imx.git

offset high_bus_count+1 when m4 is enabled

Signed-off-by: Teo Hall <teo.hall@freescale.com>
(cherry picked from commit 58983b6522c324affdbbeaa5b7b192a673c615a7)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/busfreq-imx.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-imx/busfreq-imx.c b/arch/arm/mach-imx/busfreq-imx.c
index 885ae11..e6fb748 100644
--- a/arch/arm/mach-imx/busfreq-imx.c
+++ b/arch/arm/mach-imx/busfreq-imx.c
@@ -1150,6 +1150,9 @@ static int busfreq_probe(struct platform_device *pdev)
 			high_bus_count++;
 	}
 
+	if (cpu_is_imx7d() && imx_src_is_m4_enabled())
+		high_bus_count++;
+
 	if (err) {
 		dev_err(busfreq_dev, "Busfreq init of ddr controller failed\n");
 		return err;
-- 
1.7.9.5

